File: Frontend/Certiva/src/app/Models/inscripcion-dto.ts
import { RespuestaCampoDTO } from './evento-dto';

export interface CrearInscripcionDTO {
  idUsuario: number;
  idEvento: number;
  tokenQr?: string | null;
  respuestasCampos?: RespuestaCampoDTO[];
}

export interface InscripcionDTO {
  idInscripcion: number;
  estado?: string | null;
  pagoRealizado?: boolean | null;
  fechaInscripcion?: string | null;
  idUsuario?: number | null;
  idEvento?: number | null;
  tokenQr?: string | null;
  respuestasCampos?: RespuestaCampoDTO[];
}

File: Backend/api/src/main/java/com/certiva/api/DTO/InscripcionDTO.java
package com.certiva.api.DTO;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class InscripcionDTO {

    private Long idInscripcion;

    private String estado;

    private Boolean pagoRealizado;

    private LocalDateTime fechaInscripcion;

    private Long idUsuario;

    private Long idEvento;

    private String tokenQr;

    private List<RespuestaCampoDTO> respuestasCampos = new ArrayList<>();
}
